extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/DataSourceConfig.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.quarkus.reactive.pg.client.runtime;
+
+import java.util.Optional;
+import java.util.OptionalInt;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(name = "datasource", phase = ConfigPhase.RUN_TIME)
+public class DataSourceConfig {
```

Comment:
This will conflict with the Agroal datasource config.

Reply:
It's been agreed on the mailing list and #1862 that:

- the Reactive Pg Client extension should use the usual data source config props for host, port, db, credentials and max pool size (for user experience consistency)
- db client extensions should check the data source url scheme to determine if it can be used

Reply:
Yes, those extensions should not be used at the same time, so there should not be a conflict.

Reply:
@stuartwdouglas I believe @emmanuelbernard and @Sanne vetted that. Looks OK to me as a strategy. I would have reused the `quarkus.hibernate-orm.sql-load-script` config property too if it did not belong in `hibernate-orm` :(

Reply:
I disagree, this means that you cannot mix reactive and non-reactive datasources in the same app, which goes against our core message of 'unifies reactive and imperative'.

In addition to this because this extension is postgres specific you could not mix different reactive implementations either.

I disagree that this provides 'user experience consistency'. Having the same properties do different things depending on the installed extensions is not a great user experience IMHO. 

We are talking about having auto generated documentation and self documenting config and that becomes a lot harder and a lot more confusing if different properties do different things depending on the installed extension.

Reply:
Well, that's for the default datasource, right? I guess we can:

- Support multiple datasources in reactive-pg-client
- Make sure agroal does not load datasources defined for non-jdbc drivers

This would allow one application to have two datasources pointing to jdbc and reactive. Which I think is OK, since we never want a single datasource to have two different drivers. And we don't really care if two datasources point to the same DB (they will have different drivers).

About this being postgres-specific, this is going to change soon when the other async drivers are ready. 

About documentation and config, I'm not sure how to solve this, frankly, short of making both agroal and reactive-pg-client depend on config from a third unifying module. Or decide that `quarkus.datasource` is owned by agroal and blocking, while a parallel `quarkus.reactive-datasource` config is owned by this extension, and we keep the keys as close as possible.

Please weight in @emmanuelbernard and @Sanne since you were originally in the discussion that led to this.

Reply:
Support for multiple datasources has always been a goal. Via the map support. So let’s assume we have that. 
The idea was to route the data source definition to the right extension via the url prefix or the driver. 
Today that mechanism is by each extension reading the property and ignoring if it does not belong to you. Should we have a model where an extension calls other extensions like a proper router ? I don’t know. Probably no because then we would have extension dependencies problems possibly. 

So with this framework of thinking @stuartwdouglas , what is problematic with this unified approach I.e. sharing properties across extensions for consistencies?

Reply:
>I disagree, this means that you cannot mix reactive and non-reactive datasources in the same app, which goes against our core message of 'unifies reactive and imperative'.

Yes, it's a limitation that has been identified and accepted for the initial versions.

But it's been agreed that we would use the datasource url prefix to discriminate the datasource provider in the future.

>In addition to this because this extension is postgres specific you could not mix different reactive implementations either.

MySQL and SQLServer clients are in the making on the Vert.x side. When eventually they are ready, we will be able to use the url prefix as well to decide which extension to use.

>I disagree that this provides 'user experience consistency'. Having the same properties do different things depending on the installed extensions is not a great user experience IMHO.

>We are talking about having auto generated documentation and self documenting config and that becomes a lot harder and a lot more confusing if different properties do different things depending on the installed extension.

OK. I came with this implementation after receiving the opposite feedback from @Sanne @emmanuelbernard @cescoffier and @FroMage .

Can you please raise the issue to them off-list and confirm or revisit the agreement? I know you are very busy these days but I would appreciate if it wouldn't take too long as the original discussion happened almost 4 weeks ago.

Reply:
However, one thing we said was maybe have a shared module (or was it an extension) to share the config definition and not duplicate classes. 
Maybe such extension could build step publish the data source definition and be listened to by agroal and the reactive driver

Reply:
Yes that's the third module solution I mentioned.

Reply:
@stuartwdouglas so do we have an agreement that we can solve the issues by:

- introducing a new module that holds the datasource config
- make agroal and reactive-pg-client look at that config for setup

I don't want to do this if that's not enough :)

Reply:
@emmanuelbernard @FroMage the separate module extension sounds like a good idea but, from my perspective, it's far better to release early/often and refine later than to block for another month while we agree on how to do this new extension.

Reply:
OK, so the standup confirmed we want a separate module for the config. @emmanuelbernard said we can either do this within this PR, or merge this PR as it is and work on the separate module refactor in another PR. What do you prefer?

Reply:
When is the next release? Ideally this PR would be merged along with the (updated) quickstart and guide PRs. Then I can start working on the separate module for config. From a user standpoint it wouldn't change anything (same configuration properties used).

Reply:
13th May. I think you're right it would not change anything for users. So I'll go ahead and merge this.